src/Microsoft.TestPlatform.AdapterUtilities/ManagedNameUtilities/ManagedNameHelper.Reflection.cs

```diff
@@ -266,14 +268,169 @@ private static void AppendTypeString(StringBuilder b, Type type, bool closedType
             }
         }
 
-        private static void AppendNestedTypeName(StringBuilder b, Type type)
+        private static void AppendNamespace(StringBuilder b, string ns)
```

Comment:
Some documentation explaining the business logic of this method would be nice since it doesn't do trivial work.

Unit tests would be nice too.

Reply:
These methods are tested :)